contracts/Multiplexer.sol

```diff
@@ -0,0 +1,316 @@
+///         _;
+///     }
+///
+///     function setBursar(addres newBursar)
+///         public
+///         onlyOwner()
+///     {
+///         bursar = newBursar;
+///     }
+///
+///     function disburse(address beneficiary)
+///         public
+///         onlyBursar()
+///     {
+///         beneficiary.send(1 ether);
+///     }
+/// }
+/// ```
+///
+/// But now our project is growing and our singly bursar can not handle all the
+/// work anymore. We need to have multiple bursars. We would have to change the
+/// contract and turn our `address bursar` into a `address[]` list, or a
+/// `mapping(address => bool)`.
+///
+/// This is a problem. We can not re-deploy the contract, the code would
+/// be much more complex, etc.
+///
+/// With the `Multiplexer` contract this is easy to solve. We simply deploy
+///
+/// ```
+/// multiplexer = new Multiplexer(hotWallet, [bursar_1, bursar_2, …]);
+/// hotWallet.setBursar(multiplexer);
+/// multiHotWallet = HotWallet(multiplexer);
+/// ```
+///
+/// Now all our bursars can use the `multiHotWallet` address as the new
+/// `HotWallet` contract. All the functions will magically work and everything
+/// will be forwarded to the real hot wallet contract:
+///
+/// ```
+/// multiHotWallet.disburse(target);
+/// ```
+///
+/// This does not have to be done as a retrofit. It is perfectly fine to
+/// implement `HotWallet` as above, knowing that there will be multiple bursars
+/// and then use a `Multiplexer` from the start. This has the advantage of
+/// making the smart contract code much simpler and therefore less error prone.
+///
+/// @dev Any public or external function defined in this contract will no not be
+///      forwarded. For that reason, we prefix all public interfaces with
+///      `multiplex`. We also implement our own ownership  mechanism (as opposed
+///      to inheriting from Ownable). This allows the Multiplexer to be used in
+///      the owner role.
+contract Multiplexer {
+
+    ////////////////
+    // State
+    ////////////////
+
+    /// @dev The current owner. We can not inherit from Ownable here because
+    ///     the public functions may collide with public functions on the target
+    ///     contract that we want to forward too.
+    address private owner;
+
+    /// @dev The candidate owner. For safety reasons, we do a
+    //      transfer-accept handshake to prevent the ownership
+    //      from being lost. See `owner`.
+    address private ownerCandidate;
+
+    /// @dev The target contract is the contract that we want to
+    //      forward calls too. Only the owner can change the
+    ///     target contract. In the example, the target
+    ///     contract is the existing `HotWallet` instance.
+    address private target;
+
+    /// @dev Mapping for quick lookup if a particular address belongs to a
+    ///     manager.
+    mapping (address => bool) private isManager;
+
+    /// @dev List for enumerating all current managers.
+    address[] managersList;
```

Comment:
Maybe you should extract that into a SharedOwnership (not sure about the name) and make the multiplexer inherit from it?